crates/sui-core/src/authority/transaction_deferral.rs

```diff
@@ -29,6 +30,15 @@ impl DeferralKey {
     }
 
     pub fn new_for_consensus_round(future_round: Round, deferred_from_round: Round) -> Self {
+        if future_round < deferred_from_round {
+            debug_fatal!("future_round must be greater than or equal to deferred_from_round");
```

Comment:
it would be fine for this to be a regular `fatal!`, or even collapse the whole thing to an `assert!(future_round >= deferred_from_round)`

Our philosophy with asserts is that they should be used very liberally in any situation where it could be unsafe to continue if the assert is bad. As bad as it is for validators to panic, panicking is much better than committing bad state to the chain!